core/Network/TLS/Handshake/Server.hs

```diff
@@ -603,8 +603,16 @@ filterSortCredentials rankFun (Credentials creds) =
     let orderedPairs = sortOn fst [ (rankFun cred, cred) | cred <- creds ]
      in Credentials [ cred | (Just _, cred) <- orderedPairs ]
 
-isCredentialAllowed :: Version -> Credential -> Bool
-isCredentialAllowed ver cred = pubkey `versionCompatible` ver
+isCredentialAllowed :: Version -> [ExtensionRaw] -> Credential -> Bool
+isCredentialAllowed ver exts cred =
+    -- ECDSA keys are tested against supported elliptic curves until TLS12 but
+    -- not after.  With TLS13, the curve is linked to the signature algorithm
+    -- and client support is tested with signatureCompatible13.
+    let p | ver > TLS12 = const True
```

Comment:
The comment talks about < TLS12 first but the code handles TLS13 first. I would like to have the same ordering. And I don't know why both `let` and `where` are used. So, I would like to recommend the following code:

```haskell
isCredentialAllowed :: Version -> [ExtensionRaw] -> Credential -> Bool
isCredentialAllowed ver exts cred =
    pubkey `versionCompatible` ver && satisfiesEcPredicate p pubkey
  where (pubkey, _) = credentialPublicPrivateKeys cred
        -- ECDSA keys are tested against supported elliptic curves until
        -- TLS12 but not after.  With TLS13, the curve is linked to the
        -- signature algorithm and client support is tested with
        -- signatureCompatible13.
        p | ver <= TLS12 = case extensionLookup extensionID_NegotiatedGroups exts >>= extensionDecode MsgTClientHello of
              Nothing                    -> const True
              Just (NegotiatedGroups sg) -> (`elem` sg)
          | otherwise    = const True
```

Reply:
Changed in 280bb813.